akka-stream-tests/src/test/scala/akka/stream/actor/ActorPublisherSpec.scala

```diff
@@ -262,12 +256,9 @@ class ActorPublisherSpec extends AkkaSpec with ImplicitSender {
       val source = Source[Int](senderProps)
       val sink = Sink[String](receiverProps(probe.ref))
 
-      val mat = source.collect {
+      val (snd, rcv) = source.collect {
         case n if n % 2 == 0 ⇒ "elem-" + n
-      }.to(sink).run()
-
-      val snd = mat.get(source)
-      val rcv = mat.get(sink)
+      }.toMat(sink)(Keep.both).run()
```

Comment:
`Pair.apply _` will allocate an instance for every use, `Keep.both` is preallocated